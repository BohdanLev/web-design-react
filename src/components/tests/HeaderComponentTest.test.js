import React from 'react';
import Header from '../HeaderComponent';
import { shallow } from 'enzyme';


describe('HeaderComponent', () => {
  
    const props = {
        id: null,
        username: "",
        first_name: "",
        last_name: "",
        last_login: null,
        date_joined: null,
        fetched: false,
        image: null
      };
    

      describe('Header container initial', () => {
        const headerContainer = (<Header {...props} />)

        it('renders properly', () => {
            expect(headerContainer).toMatchSnapshot()
          })    

        const component = shallow(<Header {...props} />)
        
          it('firstly no user data', ()=> {
            expect(component.state().id).toEqual(null);
            expect(component.state().first_name).toEqual("");
            expect(component.state().last_name).toEqual("");
            expect(component.state().last_login).toEqual(null);
            expect(component.state().date_joined).toEqual(null);
            expect(component.state().image).toEqual(null);
        })
     })

     

})