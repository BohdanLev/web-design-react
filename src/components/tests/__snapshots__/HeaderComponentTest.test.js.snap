// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent Header container initial renders properly 1`] = `
<Header
  date_joined={null}
  fetched={false}
  first_name=""
  id={null}
  image={null}
  last_login={null}
  last_name=""
  username=""
/>
`;
